### PR TITLE
ZCS-12510:WCAG | Classic UI | Medium Complexity | Aria issues Part - 6

### PR DESCRIPTION
**Pull request by sakshirawat0247**
1.3.1_Landmarks are not defined properly.docx

Landmarks have not been implemented for Calendar verticals toolbar buttons day, week, workweek and list button, it is considered to be out of scope as of now.It also includes the 1.3.1_landmark issue-Mail.docx.

4.1.2_aria-selected attribute wont support for div.docx

Removed the previous implemented which included defining these buttons role as "tab", added role as button.

4.1.2_name issue list text-Preferences.docx

Fixed this issue in preferences vertical for Accounts table, added required caption

See the changes of zm-ajax https://github.com/Zimbra/zm-ajax/pull/109

### CODE REVIEW

**Review by miteshsavani810 — APPROVED** [review]

File: WebRoot/js/zimbraMail/mail/view/prefs/ZmAccountsPage.js [hunk 1]
```diff
@@ -3411,19 +3414,28 @@ function(buffer, i, item, field, col, params) {
...
 		return i;
 	}
 	if (field == ZmItem.F_EMAIL) {
         var email = item.getEmail();
         var identity = item.getIdentity();
-        if (!item.isMain && identity.sendFromAddressType == ZmSetting.SEND_ON_BEHALF_OF) email = appCtxt.getActiveAccount().name + " " + ZmMsg.sendOnBehalfOf + " " + email;
-		buffer[i++] = "<div style='margin-left: 10px;'>"+ AjxStringUtil.htmlEncode(email) +"</div>";
+        if (!item.isMain && identity.sendFromAddressType == ZmSetting.SEND_ON_BEHALF_OF){
+			email = appCtxt.getActiveAccount().name + " " + ZmMsg.sendOnBehalfOf + " " + email;
+		}
```
[the remark this answers is not in the document]
Comment (sakshirawat0247) [reply WebRoot/js/zimbraMail/mail/view/prefs/ZmAccountsPage.js:3429]: fixed

**Review by miteshsavani810 — APPROVED** [review]

**Review by lavegupta — APPROVED** [review]

File: WebRoot/js/zimbraMail/abook/view/ZmContactsBaseView.js [hunk 2]
```diff
@@ -292,6 +292,11 @@ ZmContactAlphabetBar = function(parent) {
 	this._createHtml();
 
 	this._all = this._current = document.getElementById(this._alphabetBarId).rows[0].cells[0];
+	var firstRow = document.getElementById(this._alphabetBarId).rows[0];
+	if (firstRow){
+		firstRow.setAttribute('role','navigation');
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/abook/view/ZmContactsBaseView.js:297]:
```suggestion
		firstRow.setAttribute('role', 'navigation');
```

File: WebRoot/js/zimbraMail/abook/view/ZmContactsBaseView.js [hunk 3]
```diff
@@ -292,6 +292,11 @@ ZmContactAlphabetBar = function(parent) {
 	this._createHtml();
 
 	this._all = this._current = document.getElementById(this._alphabetBarId).rows[0].cells[0];
+	var firstRow = document.getElementById(this._alphabetBarId).rows[0];
+	if (firstRow){
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/abook/view/ZmContactsBaseView.js:296]:
```suggestion
	if (firstRow) {
```

File: WebRoot/js/zimbraMail/core/ZmZimbraMail.js [hunk 4]
```diff
@@ -1846,9 +1846,15 @@ function() {
 	var rootTg = appCtxt.getRootTabGroup();
 
 	var logoAnchor = document.querySelector('#skin_container_logo a');
+	var applicationrow = document.querySelector('#skin_tr_app');
 	if (logoAnchor) {
 		rootTg.addMember(logoAnchor);
 	}
+	// Add role for top row navigation of verticals	
+	if (applicationrow){
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/core/ZmZimbraMail.js:1854]:
```suggestion
	if (applicationrow) {
```

File: WebRoot/js/zimbraMail/share/view/ZmOverview.js [hunk 5]
```diff
@@ -108,8 +108,11 @@ ZmOverview = function(params, controller) {
 		var params = {container:container, threshold:15, amount:5, interval:10, id:this._scrollableContainerId};
 		this._dndScrollCallback = new AjxCallback(null, DwtControl._dndScrollCallback, [params]);
 	}
-
-	this.setAttribute('aria-label', ZmMsg.overviewLabel);
+	if (this.overviewId === 'main_Options'){
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/share/view/ZmOverview.js:111]:
```suggestion
	if (this.overviewId === 'main_Options') {
```

File: WebRoot/js/zimbraMail/abook/view/ZmContactsBaseView.js (hunk 2 above)
Comment (sakshirawat0247) [reply WebRoot/js/zimbraMail/abook/view/ZmContactsBaseView.js]: done

File: WebRoot/js/zimbraMail/abook/view/ZmContactsBaseView.js (hunk 3 above)
Comment (sakshirawat0247) [reply WebRoot/js/zimbraMail/abook/view/ZmContactsBaseView.js]: done

File: WebRoot/js/zimbraMail/core/ZmZimbraMail.js (hunk 4 above)
Comment (sakshirawat0247) [reply WebRoot/js/zimbraMail/core/ZmZimbraMail.js]: done

File: WebRoot/js/zimbraMail/share/view/ZmOverview.js (hunk 5 above)
Comment (sakshirawat0247) [reply WebRoot/js/zimbraMail/share/view/ZmOverview.js]: done

**Review by silentsakky — APPROVED** [review]

**Review by miteshsavani810 — APPROVED** [review]

**Review by silentsakky — APPROVED** [review]

**Pull request merged**